src/concrete_convex_hull.jl

```diff
@@ -1,21 +1,57 @@
+function default_convex_hull_algorithm(points)
+    if length(points) != 0 && length(first(points)) == 2
+        return "monotone_chain"
+    else
+        return nothing
+    end
+end
+
 """
-    convex_hull(points::Vector{VN}; [algorithm]::String="monotone_chain"
-               )::Vector{VN} where {N<:Real, VN<:AbstractVector{N}}
+    convex_hull(points::Vector{VN};
+                [algorithm]=default_convex_hull_algorithm(points),
+                [backend]=nothing
+                )::Vector{VN} where {N<:Real, VN<:AbstractVector{N}}
 
-Compute the convex hull of points in the plane.
+Compute the convex hull of the given points.
 
 ### Input
 
-- `points`    -- list of 2D vectors
+- `points`    -- list of vectors
 - `algorithm` -- (optional, default: `"monotone_chain"`) the convex hull
-                 algorithm, valid options are:
-
-    * `"monotone_chain"`
-    * `"monotone_chain_sorted"`
+                 algorithm, see valid options below
+- `backend`   -- (optional, default: `"nothing"`) polyhedral computation backend
+                 for higher-dimensional point sets
 
 ### Output
 
-The convex hull as a list of 2D vectors with the coordinates of the points.
+The convex hull as a list of vectors with the coordinates of the points.
+
+### Algorithm
+
+A pre-processing step treats the cases with `0`, `1` and `2` points in any dimension.
+For more than `3` points, the algorithm used depends on the dimension.
+
+For the one-dimensional case we sort the points and return the minimum and maximum points,
+in that order.
+
+The two-dimensional case is handled with a planar convex hull algorithm. The
+following algorithms are available:
+
+* `"monotone_chain"`        -- compute the convex hull of points in the plane
+                               using Andrew's monotone chain method
+* `"monotone_chain_sorted"` -- the same as `"monotone_chain"` but assuming that
+                               the points are already sorted in counter-clockwise
+                               fashion
+
+See the reference docstring of each of those algorithms for details.
+
+The higher dimensional case is treated using the concrete polyhedra library
+`Polyhedra`, that gives access to libraries such as `CDDLib` and `ConvexHull.jl`.
+These libraries can be chosen by the setting the `backend` argument.
```

Comment:
I doubted about this. But in other places we use `backend` for this and i think it's better to keep kwargs consistent to avoid confusions. Besides the same `backend` could have different algorithms.

Reply:
Agreed.